daemon/cmd/status.go

```diff
@@ -793,6 +794,42 @@ func (d *Daemon) startStatusCollector(cleaner *daemonCleanup) {
 				}
 			},
 		},
+		{
+			Name: "config-settings",
+			Probe: func(ctx context.Context) (interface{}, error) {
+				if !d.clientset.IsEnabled() || option.Config == nil {
+					return nil, fmt.Errorf("preconditions failed: no clientset or settings")
+				}
+
+				return newSettingsDelta(ctx, d)
+			},
+			OnStatusUpdate: func(status status.Status) {
+				d.statusCollectMutex.Lock()
+				defer d.statusCollectMutex.Unlock()
+
+				if status.Err != nil {
+					d.statusResponse.ConfigSettings = &models.ConfigSettings{
+						Message: status.Err.Error(),
+					}
+					return
+				}
+				sd, ok := status.Data.(settingsDelta)
+				if !ok {
+					log.Errorf("settings status update failed. expecting settingsDelta but got %T", status.Data)
+					return
+				}
+				if sd.hasDeltas() {
+					d.statusResponse.ConfigSettings = &models.ConfigSettings{
+						Deltas:  sd.deltas,
+						Message: "Agent configuration state differ from desired state. Agent needs restart",
+					}
+				} else {
+					d.statusResponse.ConfigSettings = &models.ConfigSettings{Message: "Ok"}
+				}
+				metrics.ConfigSettingsMismatch.WithLabelValues(sd.md5).Set(float64(sd.deltaCount()))
```

Comment:
Is it possible to include the config source as a label?

Reply:
Not sure what this would achieve? 
Delta counts could be from either or both source of truth i.e cm and/or crd. 
Would not really add much value imho.
If we need to clue users in on which source changed, might be best to include the info in the status output.
This will require a much heavier lift to tag the kv updates with the source origin.

Reply:
Somewhat related: How does the feature of this PR interact with [per-node configuration](https://docs.cilium.io/en/stable/configuration/per-node-config/)? In those cases the user has very much intended that a option in the configmap is overwritten with a `CiliumNodeConfig` option.

Reply:
@gandro The source of truth comes from either the cilium-configmap or the ciliumnodeconfig crd. We perform deltas to see if the current agent config differ these sources. 
Does this make sense or did I miss your question?

Reply:
Yeah after I posted the question (which was about per-node configs) I realized that I misread the code and thus deleted it :sweat_smile: Thanks for clarifying!

Reply:
@chancez Makes sense. How common is it for users to use per-node-configs on their clusters?

Reply:
I don't have that information.